api/filter/network/http_connection_manager.proto

```diff
@@ -80,6 +80,24 @@ message HttpConnectionManager {
     // populate the tag name, and the header value is used to populate the tag value. The tag is
     // created if the specified header name is present in the request's headers.
     repeated string request_headers_for_tags = 2;
+
+    // Global target percentage of requests that will be force traced if the *x-client-trace-id*
+    // header is set. Percent is resolved to the nearest 1% (rounded down). Defaults to 1 (i.e.,
+    // 100%). This variable is a direct analog for the variable of the same name in the :ref:`HTTP
+    // Connection Manager <config_http_conn_man_runtime>`.
+    Percent client_enabled = 3;
+
+    // Global target percentage of requests that will be traced after all other checks have been
+    // applied (force tracing, sampling, etc.). Percent is resolved to the nearest 1% (rounded
+    // down). Defaults to 1 (i.e., 100%). This variable is a direct analog for the variable of the
+    // same name in the :ref:`HTTP Connection Manager <config_http_conn_man_runtime>`.
+    Percent global_enabled = 4;
+
+    // Global target percentage of requests that will be randomly traced. Percent is resolved to the
+    // nearest 0.01%, rounded down. Defaults to 1 (i.e., 100%). This variable is a direct analog for
```

Comment:
We really should get out of the rounding business. If I am running a memcache service with millions of requests per second, why do I have to follow such policy?

Reply:
The API docs are telling you what Envoy itself is going to do. We need to explain to users of the API what the effect of setting a specific value will have, so somewhere we need to be clear about how fine grained percentages/ratios will be interpreted in practice.